themes/bootstrap3/js/facets.js

```diff
@@ -1,87 +1,143 @@
 /*global VuFind */
 /*exported initFacetTree */
-function buildFacetNodes(data, currentPath, allowExclude, excludeTitle, counts)
+function buildFacetNodes(facetName, data, currentPath, allowExclude, excludeTitle, showCounts, counter, locale)
 {
-  var json = [];
-  var selected = VuFind.translate('Selected');
-  var separator = VuFind.translate('number_thousands_separator');
-
-  for (var i = 0; i < data.length; i++) {
-    var facet = data[i];
-
-    var url = currentPath + facet.href;
-    var excludeUrl = currentPath + facet.exclude;
-    var item = document.createElement('span');
-    item.className = 'text';
-    if (facet.isApplied) {
-      item.className += ' applied';
+  // Helper function to create elements
+  // #todo: abstract to VuFind.el?
+  function el(tagName, className = null) {
```

Comment:
I was considering abstracting this to `VuFind.el` and modeling it [closer to this el utility](https://thescottyjam.github.io/snap.js/#!/utils/el). What do you think?

Reply:
That's a nice one! But I'd do that separately from here and apply it in all code we have (maybe apart from this piece that's just a stopgap).

Reply:
I guess the real question is how to split up the work between releases 9 and 10. We don't want to do major refactoring before 10, but maybe we could introduce the helper and start to use it in new code, and then open another PR to do bigger refactoring targeted at release 10.

There's also the question of how to group things together -- do we want to create a util or helper module to contain this type of thing? Or maybe an html module? There might be some other existing global functions that could be relocated. (I'm not sure about that, but a review of places like common.js might reveal some candidates).

Reply:
Bearing in mind that this new code would be removed in the refactoring in release 10 by #3046, it would leave us with an unused utility if #0346 goes through. That's why I'm trying to keep things contained here. If it was introduced now in other code, I'd be happy to take advantage of it here, but otherwise I'd keep it separate.

Reply:
Agreed, it wouldn't make sense to create an orphaned utility just for this. So unless we want to create a refactor targeted for 10 and then backport something from that to here, it's probably easiest to simply wait for the moment (and maybe remove the TODO comment, if it's also just going to get deleted before any action is taken).